push_new_pricing_to_statusdb.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python
+"""
+Reads the cost_calculator excel sheet and puts all that information
+into statusdb.
+"""
+import argparse
+from openpyxl import load_workbook
+import coloredlogs
+import logging
+import yaml
+from couchdb import Server
+import datetime
+from collections import OrderedDict
+
+FIRST_ROW = {'components': 9,
+             'products': 4}
+SHEET = {'components': 'Price list',
+         'products': 'Products'}
+
+# Skip columns which are calculated from the other fields
+SKIP = {'components': ['Price', 'Total', 'Per unit'],
+        'products': ['Internal', 'External']}
+
+# The name of the _id_ key and which variables that cannot be changed
+# while keeping the same _id_. If an update of any of these fields is needed,
+# a new id needs to be created.
+CONSERVED_KEY_SETS = {'products': ('ID', ['Category', 'Type', 'Name']),
+                      'components': ('REF_ID', ['Category', 'Type', 'Product name'])}
+
+# The combination of these "columns" need to be unique within the document
+UNIQUE_KEY_SETS = {'products': ('ID', ['Category', 'Type', 'Name']),
+                   'components': ('REF_ID', ['Category', 'Type', 'Product name', 'Units'])}
+
+MAX_NR_ROWS = 200
+
+# Assuming the rows of products are sorted in the preferred order
+
+# Set up a logger with colored output
+logger = logging.getLogger(__name__)
+logger.propagate = False  # Otherwise the messages appeared twice
+coloredlogs.install(level='INFO', logger=logger,
+                    fmt='%(asctime)s %(levelname)s %(message)s')
+
+
+def check_unique(items, type):
+    """Make sure all items within _items_
+
+    fulfill the uniqueness criteria according to the UNIQUE_KEY_SETS
+    """
+    key_val_set = set()
+    for id, item in items.items():
+        id_key, keys = UNIQUE_KEY_SETS[type]
+        t = tuple(item[key] for key in keys)
+
+        # Check that it is not already added
+        if t in key_val_set:
+            raise ValueError("Key combination {}:{} is included multiple "
+                             "times in the {} sheet. "
+                             "ABORTING.".format(keys, t, type))
+        key_val_set.add(t)
+    return True
+
+
+def check_conserved(new_items, current_items, type):
+    """Ensures the keys in CONSERVED_KEY_SETS are conserved for each given id.
+
+    Compares the new version against the currently active one.
+    Params:
+        new_items     - A dict of the items that are to be added
+                        with ID attribute as the key.
+        current_items - A dict of the items currently in the database
+                        with ID attribute as the key.
+        type          - Either "components" or "products"
+    """
+    conserved_keys = CONSERVED_KEY_SETS[type][1]
+
+    for id, new_item in new_items.items():
+        if str(id) in current_items:
+            for conserved_key in conserved_keys:
+                if new_item[conserved_key] != current_items[str(id)][conserved_key]:
```

Comment:
That could happen if the excel sheet doesn't have all of the conserved columns. But it would lead to the behaviour we would like to since we would like the script to abort in this case. But I'll change it to give a more easily understood error message. Thanks!